### PR TITLE
fix(api): Adds per-axis homing speeds to OT3Config separate from motion-settings

### PR DESCRIPTION
**Pull request by andySigler**
# Overview

(not sure if this implementation vibes with what is ideal/expected from SW)

Adds per-axis speeds to use for calls to `OT3Controller.home()`, where the homing speeds are defined as separate entities from the per-axis discontinuities set in `OT3Config.motion_settings`.

The issues with the prior implementation was that changing an axis' max-discontinuity would directly affect that axis' homing speeds. So, for example, if I wanted to make the XY gantry movements smoother to reduce water sloshing, I could reduce the max-discontinuity to a small number. BUT, this would then make the homing speed super slow.

# Changelog

# Review requests

# Risk assessment

### CODE REVIEW

**Review by sfoster1 — CHANGES_REQUESTED** [review]
As I understand the problem here, I think this is the wrong way to solve it.

We really want to have core motion configuration that is accurate to the physical limits of the robot, because there will always be times that we want to use them - even if you want to reduce sloshing, you presumably have a lot of moves where you're not carrying a container full of liquid where you also don't want to be super slow.

What I'd advocate for instead of this change is the ability to select, call them "motion contexts", from higher levels of the code. This is similar to the gcode concept of [travel move vs tool move](https://marlinfw.org/docs/gcode/G000-G001.html). You can see it as an extension of the ability to pass a maximum speed to `move()`, and it could be implemented with a token concept:
- have a couple well-known motion contexts for things like "travel", "carrying liquid", etc
    - these could live in robot config and show up in default config
    - they likely would need versions for different gantry configurations
- have a way to add some new ones temporarily
- this way looks like `def create_move_context(self, name: str, settings: SomeSettingsType) -> str: ...`
- then you make `move()` take a `motion_context_name` param that is a string that is either a well-known motion context name or the name of a previously created user context
- the expectation is that well-known motion parameters would not change (or would change in a versioned manner) and that user motion contexts would be created in a protocol and be at least removed on reboot if not explicitly removed by the protocol middleware

tl;dr: let's not do this because i think there's a really good different way to solve the problem.

**Comment by sfoster1** [comment]
I'm really glad you looked into this by the way.
